source/adios2/toolkit/format/buffer/heap/BufferSTL.tcc

```diff
@@ -0,0 +1,51 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * BufferSTL.tcc
+ *
+ *  Created on: Sep 18, 2019
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#ifndef ADIOS2_TOOLKIT_FORMAT_BUFFER_HEAP_BUFFERSTL_TCC_
+#define ADIOS2_TOOLKIT_FORMAT_BUFFER_HEAP_BUFFERSTL_TCC_
+
+#include "BufferSTL.h"
+
+#include <memory>
+
+namespace adios2
+{
+namespace format
+{
+
+template <class T>
+size_t BufferSTL::Align() const noexcept
+{
+    auto lf_align = [](const size_t align, const size_t size, void *&ptr,
+                       size_t &space) {
+        const uintptr_t largePtr = reinterpret_cast<uintptr_t>(ptr);
+        const uintptr_t aligned = (largePtr - 1u + align) & -align;
+        const uintptr_t diff = aligned - largePtr;
+        if ((size + diff) <= space)
+        {
+            space -= diff;
+        }
+    };
```

Comment:
I'm a bit concerned with the implementation as it seems to be pulled from or at least derived from the implementation in libstdc++, which is gpl or lgpl.  Can we grab the implementation directly from the llvm libc++ as to avoid any license issues?

Reply:
will do. Thanks for the reminder on licenses.

Reply:
Moved to the llvm implementation which is MIT licensed.

Reply:
We just need a comment to note attribution.  Something like this should be sufficient:
```
// std::align implementation from llvm libc++
```
Thanks for making the change